spec/src/main/resources/META-INF/smithy/bsp/extensions/rust.smithy

```diff
@@ -0,0 +1,393 @@
+}
+
+@enumKind("closed")
+intEnum RustTargetKind {
+    /// For lib targets.
+    LIB = 1
+    /// For binaries.
+    BIN = 2
+    /// For integration tests.
+    TEST = 3
+    /// For examples.
+    EXAMPLE = 4
+    /// For benchmarks.
+    BENCH = 5
+    /// For build scripts.
+    CUSTOMBUILD = 6
+    /// For unknown targets.
+    UNKNOWN = 7
+}
+
+list RustPackageRequiredFeatures {
+    member: String
+}
+
+list RustFeatures {
+    member: RustFeature
+}
+
+structure RustFeature {
+    /// Name of the feature.
+    @required
+    name: String
+    /// Feature's dependencies.
+    @required
+    dependencies: RustFeatureDependencies
+}
+
+list RustFeatureDependencies {
+    member: String
+}
+
+structure RustCfgOptions {
+    /// `cfgs` in Rust can take one of two forms: "cfg1" or "cfg2=\"string\"".
+    /// The `cfg` is split by '=' delimiter and the first half becomes key and
+    /// the second is aggregated to the value in `keyValueOptions`.
+    keyValueOptions: KeyValueOptions
+    /// A sequence of first halves after splitting `cfgs` by '='.
+    nameOptions: RustCfgOptionsNames
+}
+
+list RustCfgOptionsNames {
+    member: String
+}
+
+map KeyValueOptions {
+    key: String
+    value: Values
+}
+
+list Values {
+    member: String
+}
+
+map RustEnvironmentVariables {
+    key: String
+    value: String
+}
+
+map RustRawDependencies {
+    /// Package id
+    key: String
+    value: RustRawDependenciesInfo
+}
+
+list RustRawDependenciesInfo {
+    member: RustRawDependency
+}
+
+structure RustRawDependency {
+    /// The name of the dependency.
+    @required
+    name: String
+    /// Name to which this dependency is renamed when declared in Cargo.toml.
+    rename: String
+    /// The dependency kind. "dev", "build", or null for a normal dependency.
+    kind: String
+    /// The target platform for the dependency.
+    target: String
+    /// Indicates whether this is an optional dependency.
+    @required
+    optional: Boolean
+    /// Indicates whether default features are enabled.
+    @required
+    usesDefaultFeatures: Boolean
+    /// A sequence of enabled features.
+    @required
+    features: RustRawDependencyFeatures
+}
+
+list RustRawDependencyFeatures {
+    member: String
+}
+
+map RustDependencies {
+    /// Package id
+    key: String
+    value: RustDependenciesInfo
+}
+
+list RustDependenciesInfo {
+    member: RustDependency
+}
+
+structure RustDependency {
+    /// The Package ID of the dependency.
+    @required
+    pkg: String
+    /// The name of the dependency's library target.
+    /// If this is a renamed dependency, this is the new name.
+    name: String
+    /// Array of dependency kinds.
+    depKinds: RustDepKindsInfo
+}
+
+list RustDepKindsInfo {
+    member: RustDepKindInfo
+}
+
+structure RustDepKindInfo {
+    /// The dependency kind.
+    @required
+    kind: RustDepKind
+    /// The target platform for the dependency.
+    target: String
+}
+
+@enumKind("closed")
+intEnum RustDepKind {
```

Comment:
is it the same as https://github.com/build-server-protocol/build-server-protocol/pull/557/files#diff-6e11ce263f60ffd4930e220448c2eb03ecbcd309a91336165f5fea75bc21ed4cR259 ?

Reply:
Yeah, we can use it there